Plutarch/PLam.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Plutarch.PLam (
```

Comment:
Do you intend for this module to be exposed to end-users?

Reply:
Nope; should it be put under `Plutarch.Internal.PLam`? 

(We also have *another* Internal, at `Plutarch.DataRepr.Internal`, which can be a bit confusing).

Reply:
We have a lot of internals! Maybe it should all be under `Plutarch/Internal`

Reply:
Moved to `Plutarch.Internal.PLam`. I didn't touch the DataRepr stuff because it is outside the scope of this PR.

Reply:
I think `PlutusType` should be made internal too? It is not meant to be exported directly either. We split these modules only because Plutarch.hs was growing to be big.

Reply:
> I think `PlutusType` should be made internal too? It is not meant to be exported directly either. We split these modules only because Plutarch.hs was growing to be big.

Done now.